src/profiles-table-layout.jl

```diff
@@ -12,15 +12,31 @@ a `key = value` list of parameters. Explicit keyword arguments take precedence.
 
 ## Parameters
 
-- `value::Symbol = :value`: The column name with the profile values.
-- `timestep::Symbol = :timestep`: The column name with the time steps in the profile.
-- `period::Symbol = :period`: The column name with the period number in the profile.
+- `value::Symbol = :value`: The column name with the profile values. This is a required column.
+- `timestep::Symbol = :timestep`: The column name with the time steps in the profile. This is a required column.
+- `period::Symbol = :period`: The column name with the period number in the profile. This is a required column.
+- `year::Symbol = :year`: The column name with the year of the profile. This is an optional column.
+  If not present, all profiles are assumed to be from `default_year`.
+- `default_year::Int = 2000`: The default year to use if the year column is not present
+  in the profiles.
+- `scenario::Symbol = :scenario`: The column name with the scenario of the profile. This is an optional column.
+  If not present, all profiles are assumed to be from `default_scenario`.
+- `default_scenario::Int = 1`: The default scenario to use if the scenario column is
+  not present in the profiles.
+- `cols_to_groupby::Vector{Symbol} = [:year]`: The column names to group by when
+  performing clustering on groups of profiles separately. If empty, no grouping is done. Default is to group by `year`.
+
 """
 Base.@kwdef struct ProfilesTableLayout
   value::Symbol = :value
   timestep::Symbol = :timestep
   period::Symbol = :period
-  cols_to_groupby::Vector{Symbol} = Symbol[]
+  profile_name::Symbol = :profile_name
+  year::Symbol = :year
+  default_year::Int = 2000
+  scenario::Symbol = :scenario
+  default_scenario::Int = 1
+  cols_to_groupby::Vector{Symbol} = [:year]
```

Comment:
Apparently, this magic works to get the correct year Symbol to `cols_to_groupby`:

```suggestion
  cols_to_groupby::Vector{Symbol} = [year]
```

So there is no need for a specific constructor anymore